src/components/MeetingsPage/MeetingsPage.jsx

```diff
@@ -0,0 +1,222 @@
+  const portalManager = usePortals();
+  const [meetingToAdd, setMeetingToAdd] = useState({ 
+    patientName: '', 
+    doctorName: '', 
+    date: '', 
+    reports: '' 
+  });
+  const [meetings, setMeetings] = useState([]);
+  const [isDisabled, setIsDisabled] = useState(true);
+  const [isSnackbarOpened, setIsSnackbarOpened] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const showSnackbar = (message) => {
+    setIsSnackbarOpened(true);
+    setErrorMessage(message);
+  };
+
+  const handleChange = (key, value) => {
+    setMeetingToAdd({...meetingToAdd, [key]: value});
+  }
+
+  const getMeetings = async () => {
+    try {
+      const resp = await store.getAllMeetings();
+      if (resp.statusText === 'OK') {
+        setMeetings(resp.data);
+      }
+    } catch (error) {
+      showSnackbar('Невозможно получить приемы!');
+    }
+  }
+
+  const ifEmpty = () => {
+    if (meetingToAdd.patientName === ''
+      || meetingToAdd.doctorName === ''
+      || meetingToAdd.date === ''
+      || meetingToAdd.reports === ''
+    ) {
+      return setIsDisabled(true);
+    }
+    return setIsDisabled(false);
+  }
+
+  const clearPatientInfo = () => {
+    setMeetingToAdd({...meetingToAdd, patientName: '', doctorName: '', date: '', reports: '' });
+  }
+
+  const addNewMeeting = async () => {
+    try {
+      const resp = await addMeeting(meetingToAdd);
+      if (resp.statusText === 'OK') {
+        setMeetings([...meetings, resp.data]);
+        clearPatientInfo();
+      }
+    } catch (error) {
+      showSnackbar('Невозможно добавить прием!');
+    }
+  }
+
+    const deleteOneMeeting = async (id) => {
+    try {
+      const resp = await deleteMeeting(id);
+      if (resp.statusText === 'OK') {
+        setMeetings(meetings.filter(meeting => meeting._id !== id));
+      }
+    } catch (error) {
+      showSnackbar('Невозможно удалить прием!');
+    }
+  }
+
+  const editOneMeeting = async (meeting) => {
+    try {
+        const resp = await editMeeting(meeting);
+        if (resp.statusText === 'OK') {
+          setMeetings(meetings.map(visit => {
+            if (visit._id === meeting._id) {
+              visit = resp.data;
+            }
+            return visit;
+          }))
+        }
+      } catch (error) {
+        showSnackbar('Невозможно изменить прием!');
+      }
+    }
+
+    const showConfirmDelete = (id) => {
+      portalManager.open(
+        portal => <ConfirmDelete 
+          closeConfirmDelete={portal.close} 
+          deleteOneMeeting={deleteOneMeeting}
+          id={id}
+        />
+      );
+    }
+
+    const showConfirmEdit = (meeting) => {
+    portalManager.open(
+      portal => <ConfirmEdit
+        closeConfirmEdit={portal.close} 
+        meeting={meeting}
+        editOneMeeting={editOneMeeting}
+      />
+    );
+  }
+
+  useEffect(() => {
+    if (meetings.length === 0) {
+      getMeetings();
+    }
+    ifEmpty();
+  }, [meetingToAdd]);
+
+  return (
```

Comment:
divide this component into more components. Let this page remain for assembling all the components. But you need to take out the form for adding appointments to a separate component. And a table of all entries on the reception in another separate component